src/libraries/Microsoft.Extensions.DependencyInjection/tests/DI.Tests/ActivatorUtilitiesTests.cs

```diff
@@ -0,0 +1,208 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using Xunit;
+
+namespace Microsoft.Extensions.DependencyInjection.Tests
+{
+    public class ActivatorUtilitiesTests
+    {
+        [Fact]
+        public void ShouldUseLongestAvailableConstructorOnlyIfConstructorsHaveTheSamePriority()
+        {
+            var services = new ServiceCollection();
+            services.AddScoped<B>();
+            services.AddScoped<S>();
+            using var provider = services.BuildServiceProvider();
+            var a = new A();
+            var c = new C();
+
+            var instance = ActivatorUtilities.CreateInstance<Creatable>(provider, a, c);
+
+            Assert.Null(instance.B);
```

Comment:
Why should `B` be null here? There is a scoped service for `B`, so shouldn't the ctor that takes a `B` be picked?

Reply:
In the discussion in https://github.com/dotnet/runtime/issues/46132, it is asking for an ambiguous exception to be thrown to be thrown in this case. Which seems like the right thing IMO. If there are multiple ctors that we can't really pick between, it is better to throw and say "use the `ActivatorUtilitiesConstructorAttribute` to disambiguate". It is really hard to define perfect behavior here when the "given arguments" and the "services available" can intermix.

See also all the discussion on https://github.com/dotnet/runtime/issues/46132 for all the scenarios, and the intended behaviors.

Reply:
@mapogolions - any thoughts on this?

Reply:
@eerhardt I don't see a way to satisfy all the mentioned requirements (especially ambiguity detection).  Feel free to close this as a dead end.